packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -1671,6 +1674,130 @@ class _AnimatedRotationState extends ImplicitlyAnimatedWidgetState<AnimatedRotat
+/// example, an [Offset] with a `dx` of 0.25 will result in a horizontal
+/// translation of one quarter the width of the child.
+///
+/// {@tool snippet}
+///
+/// This code defines a widget that uses [AnimatedOffset] to translate a [FlutterLogo]
+/// up or down by the amount of it's height with each press of the corresponding button.
+///
+/// ```dart
+/// class LogoTranslate extends StatefulWidget {
+///   const LogoTranslate({Key? key}) : super(key: key);
+///
+///   @override
+///   State<LogoTranslate> createState() => LogoTranslateState();
+/// }
+///
+/// class LogoTranslateState extends State<LogoTranslate> {
+///  Offset offset = Offset(0, 0);
+///
+///  void _moveUp() {
+///    setState(() => offset -= Offset(0, 1));
+///  }
+/// 
+///  void _moveDown() {
+///    setState(() => offset += Offset(0, 1));
+///  }
+///
+///  @override
+///  Widget build(BuildContext context) {
+///    return Column(
+///      mainAxisSize: MainAxisSize.min,
+///      children: <Widget>[
+///        ElevatedButton(
+///          child: const Text('Translate up'),
+///          onPressed: _moveUp,
+///        ),
+///        ElevatedButton(
+///          child: const Text('Translate down'),
+///          onPressed: _moveDown,
+///        ),
+///        Padding(
+///          padding: const EdgeInsets.all(50),
+///          child: AnimatedOffset(
+///            offset: offset,
+///            duration: const Duration(milliseconds: 500),
+///            curve: Curves.easeInOut,
+///            child: const FlutterLogo(),
+///          ),
+///        ),
+///      ],
+///    );
+///  }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [AnimatedPositioned], which, as a child of a [Stack], automatically
+///    transitions its child's position over a given duration whenever the given
+///    position changes.
+///  * [AnimatedAlign], which automatically transitions its child's
+///    position over a given duration whenever the given [alignment] changes.
+class AnimatedOffset extends ImplicitlyAnimatedWidget {
+  /// Creates a widget that animates its offset translation implicitly.
+  ///
+  /// The [offset] and [duration] arguments must not be null.
+  const AnimatedOffset({
+    Key? key,
+    this.child,
+    required this.offset,
+    Curve curve = Curves.linear,
+    required Duration duration,
+    VoidCallback? onEnd,
+  }) :
+        super(key: key, curve: curve, duration: duration, onEnd: onEnd);
+
+  /// The widget below this widget in the tree.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget? child;
+
+  /// The target offset. 
+  /// The child will be translated horizontally by `width * dx` and vertically by `height * dy`
+  ///
+  /// The offset must not be null.
+  final Offset offset;
+
+  @override
+  _AnimatedOffsetState createState() => _AnimatedOffsetState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
```

Comment:
Add the offset as a property here.

Reply:
It looks like you used a DoubleProperty which I think may fail at runtime because it expects the property value to be a double. You can test this by running a test app in debug mode and typing `w` to dump the widget hierarchy and looking for your AnimatedSlide widget.

Other locations that express an `Offset` as a property use a raw `DiagnosticsProperty` as in `properties.add(DiagnosticsProperty<Offset>('origin', origin));` (see RenderTransform).

Reply:
Actually, it doesn't require a runtime tests, the builds are failing on this line which is causing (nearly) all of the red results in the test runs.

Reply:
Thanks! I changed it now to `DiagnosticsProperty<Offset>`